### PR TITLE
Changed state to update properly, added empty views array on initialization of a new media type

### PR DESCRIPTION
**Pull request by jmourell**

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -38,17 +38,18 @@ function reducer(state = initialState, { type, data }) {
       return {
         ...state,
         activeMediaId: data.campaigns[0].activeMediaId,
-        campaigns: { ...data.campaigns }
+        campaigns: data.campaigns.slice(0)
```
Comment (wilandrade) [inline src/reducers/index.js]: Why is this better than destructuring?

File: src/reducers/index.js [hunk 2]
```diff
@@ -60,10 +61,23 @@ function reducer(state = initialState, { type, data }) {
...
+      });
+      return {
+          ...state,
+          campaigns: newCampaigns
+          
+      }
+      
+        
+      // const newMedia = { ...data };
+      // const newState = { ...state };
```
Comment (wilandrade) [inline src/reducers/index.js]: please delete comments

**Review by wilandrade — CHANGES_REQUESTED** [review]

**Comment by wilandrade** [comment]
Please remember to add descriptions to all pull requests.

Good work here!

File: src/reducers/index.js (hunk 1 above)
Comment (jmourell) [reply src/reducers/index.js]: It really depends on what you'd like to do I can keep it as an object of objects or change it to an array of objects.  I can destructure it as an array too. Either way is alright.

Comment (wilandrade) [reply src/reducers/index.js]: Ahh, I see it now. It was set up as an object...

**Review by wilandrade — APPROVED** [review]
LGTM

**Pull request merged**